src/main/java/domain/ladder/Ladder.java

```diff
@@ -0,0 +1,57 @@
+package domain.ladder;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class Ladder {
+    private final List<Line> lines;
+    private final LadderWidth width;
+    private final LadderHeight height;
+
+    public Ladder(LadderWidth width, LadderHeight height) {
+        this.width = width;
+        this.height = height;
+        this.lines = createLadder();
+    }
+
+    private List<Line> createLadder() {
+        List<Line> ladderLines = new ArrayList<>();
+        for (int i = 0; i < height.getValue(); i++) {
+            ladderLines.add(new Line(width.getValue()));
+        }
+        return ladderLines;
+    }
+
+    public int getWidth() {
+        return width.getValue();
+    }
+
+    public List<Line> getLines() {
+        return Collections.unmodifiableList(lines);
+    }
+
+    public List<String> calculateResults(List<String> gameResults) {
+
+        List<String> results = new ArrayList<>();
+
+        for (int i = 0; i < getWidth(); i++) {
+            int endPoint = calculateEndPoint(i);
+            results.add(gameResults.get(endPoint));
+        }
+        return results;
+    }
+
+    private int calculateEndPoint(int startPoint) {
+        int currentPoint = startPoint;
+
+        for (Line line : lines) {
+            if (line.isAbleToMoveRight(currentPoint)) {
+                currentPoint++;
+            } else if (line.isAbleToMoveLeft(currentPoint)) {
+                currentPoint--;
+            }
+        }
```

Comment:
리뷰를 잘 반영해주셨군요!
`currentPoint`가 이동 하는 조건을 line이 담당하면서 흐름을 쉽게 이해 할 수 있게 되었어요.

이제 depth를 줄여볼까요? + else도 제거
```suggestion
        for (Line line : lines) {
            currentPoint = moveByCurrentPoint(currentPoint)
        }
        
        private int moveByCurrentPoint(Line line, int currentPoint) {
            if (line.isAbleToMoveRight(currentPoint)) {
                return currentPoint + 1;
            }
            
            return currentPoint - 1;
        }               
```

메서드를 분리해서 depth를 줄이고, early-return으로 else 단계도 없애봤어요.
가독성이 개선 되었다고 느껴지시나요?